index.d.ts

```diff
@@ -1,12 +1,65 @@
-import { Parser } from 'acorn' 
+import * as acorn from 'acorn';
 
-declare const jsx: (options?: jsx.Options) => (BaseParser: typeof Parser) => typeof Parser;
+declare const jsx: {
+  tokTypes: typeof acorn.tokTypes;
+  (options?: jsx.Options): (BaseParser: typeof acorn.Parser) => jsx.AcornJsxParser
+}
 
 declare namespace jsx {
-  interface Options {
+
+  type tokTypes = typeof acorn.tokTypes;
+
+  export interface TokTypes extends tokTypes {
+    jsxName: acorn.TokenType,
+    jsxText: acorn.TokenType,
+    jsxTagEnd: acorn.TokenType,
+    jsxTagStart: acorn.TokenType
+  }
+
+  export interface Options {
     allowNamespacedObjects?: boolean;
     allowNamespaces?: boolean;
   }
+
+  export type TokContexts = {
+    tc_oTag: acorn.TokContext,
+    tc_cTag: acorn.TokContext,
+    tc_expr: acorn.TokContext
+  };
+
+  type P = typeof acorn.Parser;
+
+  // We pick (statics) from acorn rather than plain extending to avoid complaint
+  //   about base constructors needing the same return type (i.e., we return
+  //   `IAcornJsxParser` here)
+  export interface AcornJsxParser extends Pick<P, keyof P> {
+    readonly acornJsx: {
+      tokTypes: TokTypes;
+      tokContexts: TokContexts
+    };
+
+    new (options: acorn.Options, input: string, startPos?: number): IAcornJsxParser;
+  }
+
+  export interface IAcornJsxParser extends acorn.Parser {
```

Comment:
In Java and C# it’s often customary to prefix interfaces with `I`. In TypeScript this is mostly considered a bad practice.

I’m not saying it’s not wrong, but I think it’s best to avoid it.

```suggestion
  export interface AcornJsxParser extends acorn.Parser {
```

Reply:
Good to know the convention, thanks. Applied in latest commit.